packages/core/src/common/keybinding.spec.ts

```diff
@@ -152,43 +154,77 @@ describe('keybindings', () => {
     it("should only return the more specific keybindings when a keystroke is entered", () => {
         const rawKeybindingsUser: RawKeybinding[] = [{
             command: "test.command",
-            keybinding: "ctrl+b"
+            keybinding: "shift+b"
         }];
 
         keybindingRegistry.setKeymap(KeybindingScope.USER, rawKeybindingsUser);
 
         const rawKeybindingsSpecific: RawKeybinding[] = [{
             command: "test.command",
-            keybinding: "ctrl+c"
+            keybinding: "shift+c"
         }];
 
-        const validKeyCode = KeyCode.createKeyCode({ first: Key.KEY_C, modifiers: [Modifier.M1] });
+        const validKeyCode = KeyCode.createKeyCode({ first: Key.KEY_C, modifiers: [Modifier.M2] });
 
         keybindingRegistry.setKeymap(KeybindingScope.WORKSPACE, rawKeybindingsSpecific);
 
-        let bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_A, modifiers: [Modifier.M1] }));
+        let bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_A, modifiers: [Modifier.M2] }));
         expect(bindings).to.be.empty;
 
-        bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_B, modifiers: [Modifier.M1] }));
+        bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_B, modifiers: [Modifier.M2] }));
         expect(bindings).to.be.empty;
 
-        bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_C, modifiers: [Modifier.M1] }));
+        bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_C, modifiers: [Modifier.M2] }));
         expect(bindings[0].keyCode.key).to.be.equal(validKeyCode.key);
     });
 });
 
 describe("keys api", () => {
-    it("parses a string to a KeyCode correctly", () => {
-        let keycode = KeyCode.parse("ctrl+b");
+    it("should parse a string to a KeyCode correctly", () => {
+        let keycode = KeyCode.parse("shift+b");
         expect(keycode).is.not.undefined;
         if (keycode) {
+            expect(keycode.shift).to.be.true;
             expect(keycode.key).is.equal(Key.KEY_B);
-            expect(keycode.ctrl).to.be.true;
         }
         // Invalid keystroke string
         keycode = KeyCode.parse("ctl+b");
         expect(keycode).is.undefined;
     });
+
+    if (isOSX) {
+        it("should parse a string containing ctrl/option to a KeyCode correctly (OS X)", () => {
+            const keycodeCtrl = KeyCode.parse("ctrl+b");
+            expect(keycodeCtrl).is.not.undefined;
+            if (keycodeCtrl) {
+                expect(keycodeCtrl.meta).to.be.true;
```

Comment:
I'm not sure I understand how this works. What's the mapping here?
ctrl -> meta
option -> alt
cmd -> ?

Reply:
I think `cmd` is like `meta` here.

Reply:
So it's like this?
string representation -> key code field -> physical key
"ctrl" -> meta -> cmd
"option" -> alt -> option
"shift" ->  shift -> shift
"cmd" ->  meta -> cmd ?
It looks like we lost a key

the code here parses "ctrl" into meta, which means cmd. I understand that this string representation is what the user would write in the json file, right? How would one actually make a "ctrl+b" keybinding if it gets converted to cmd+b internally?

Reply:
Indeed I think you're right (i.e a user can't remap `ctrl` to the OS X `ctrl` as it would become a `command`). Let me try to fix it.

Reply:
I've pushed a new patch which adds tests to the cases you mentionned. Can you confirm that I understood correctly the cases you mentionned above? i.e if you bind `meta` or `cmd` they simply won't work in Windows, instead of fallback'ing  to `ctrl`, and you can use `ControlOrCommand`, `CommandOrControl` to bind something that would work for the two platform. I don't like the caps though, idk if theres better strings for this...